packages/repository/src/relations/has-one/has-one.repository.ts

```diff
@@ -87,4 +104,23 @@ export class DefaultHasOneRepository<
     }
     return found[0];
   }
+
+  async delete(where?: Where<TargetEntity>, options?: Options): Promise<Count> {
+    const targetRepository = await this.getTargetRepository();
+    return targetRepository.deleteAll(
+      constrainWhere(where, this.constraint as Where<TargetEntity>),
+      options,
+    );
+  }
+
+  async patch(
+    dataObject: DataObject<TargetEntity>,
+    options?: Options,
+  ): Promise<Count> {
+    const targetRepository = await this.getTargetRepository();
+    return await targetRepository.updateAll(
+      constrainDataObject(dataObject, this.constraint),
+      constrainWhere({}, this.constraint as Where<TargetEntity>),
```

Comment:
Similarly here, why is the explicit cast to `Where<TargetEntity>` needed? Can we find a way that would not require it?

Reply:
Ok

Reply:
@RaphaelDrai PTAL, can we remove this explicit type cast here?

Reply:
Do we need to call `constrainWhere` at all? I think we can use `this.constraint` directly.

```suggestion
      this.constraint,
```